torch/csrc/DataLoader.cpp

```diff
@@ -38,9 +44,8 @@ static void setSignalHandler(int signal, void(*handler)(int, siginfo_t *, void *
 {
   struct sigaction sa;
   sa.sa_sigaction = handler;
-  sa.sa_flags = SA_RESTART|SA_SIGINFO|SA_NOCLDSTOP;
-  sigemptyset(&sa.sa_mask);
-  if (sigaction(signal, &sa, old_sa_ptr) != 0) {
+  sa.sa_flags = SA_RESTART|SA_SIGINFO|SA_NOCLDSTOP|SA_NODEFER;
```

Comment:
what's the new flag?

Reply:
It allows the same signal to be triggered within the handler rather than being deferred after handler (http://man7.org/linux/man-pages/man2/sigaction.2.html).

Actually, upon thinking about it. It doesn't do anything for this, because raising the signal is the last thing that handler do. Should I remove it?

Reply:
Nah, that should be fine. Thanks for the description!